Source/WebCore/dom/TreeScope.cpp

```diff
@@ -239,27 +239,36 @@ Element* TreeScope::ancestorElementInThisScope(Element* element) const
 void TreeScope::addImageMap(HTMLMapElement& imageMap)
 {
     AtomStringImpl* name = imageMap.getName().impl();
-    if (!name)
+    AtomStringImpl* id = imageMap.getIdAttribute().impl();
+    if (!name && !id)
         return;
     if (!m_imageMapsByName)
         m_imageMapsByName = makeUnique<TreeScopeOrderedMap>();
-    m_imageMapsByName->add(*name, imageMap, *this);
+    if (name)
+        m_imageMapsByName->add(*name, imageMap, *this);
+    if (id)
+        m_imageMapsByName->add(*id, imageMap, *this);
 }
 
 void TreeScope::removeImageMap(HTMLMapElement& imageMap)
 {
     if (!m_imageMapsByName)
         return;
     AtomStringImpl* name = imageMap.getName().impl();
-    if (!name)
-        return;
-    m_imageMapsByName->remove(*name, imageMap);
+    AtomStringImpl* id = imageMap.getIdAttribute().impl();
+    if (name)
+        m_imageMapsByName->remove(*name, imageMap);
+    if (id)
+        m_imageMapsByName->remove(*id, imageMap);
 }
 
 HTMLMapElement* TreeScope::getImageMap(const AtomString& name) const
 {
     if (!m_imageMapsByName || !name.impl())
         return nullptr;
+    if (name.isEmpty())
```

Comment:
We no longer need the `!name.impl()` check on the previous line if you add this check.

Reply:
Thanks! Will fix it, just trying to understand why there are assertion failure on Debug build.